### PR TITLE
 [ES] Use `title.sort` only when no other sort is available, refs 3323

### PR DESCRIPTION
**Pull request by kghbln**
This PR is made in reference to: #3323 

This PR addresses or contains:
- As suggested https://github.com/SemanticMediaWiki/SemanticMediaWiki/issues/3323#issuecomment-414194051

This PR includes:
- [ ] Tests (unit/integration)
- [x] CI build passed

Fixes #

### CODE REVIEW

File: src/Elastic/QueryEngine/SortBuilder.php [hunk 1]
```diff
@@ -96,6 +96,7 @@ private function getFields( array $sortKeys ) {
 		$isConstantScore = true;
 		$sort = [];
 		$sortFields = [];
+		$count = count( $sortKeys );
```
Comment (JeroenDeDauw) [inline src/Elastic/QueryEngine/SortBuilder.php]: sortKeyCount would be a better name

File: src/Elastic/QueryEngine/SortBuilder.php [hunk 2]
```diff
@@ -117,12 +118,16 @@ private function getFields( array $sortKeys ) {
...
-	private function addDefaultField( &$sort, $order ) {
+	private function addDefaultField( &$sort, $order, $count ) {
 		$sort['subject.sortkey.sort'] = [ 'order' => $order ];
 
 		// Add title as extra criteria in case an entity uses the same sortkey
 		// to clarify its relative position, @see T:P0416#8
-		$sort['subject.title.sort'] = [ 'order' => $order ];
+		// Only add the title as determining factor when no other sort parameter
+		// is available
+		if ( $count == 1 ) {
```
Comment (JeroenDeDauw) [inline src/Elastic/QueryEngine/SortBuilder.php]: In PHP always use === or !== (unless good reason not to)

**Comment by JeroenDeDauw** [comment]
Looks good to me, though I am not up to speed with the context

**Comment by mwjames** [comment]
> however we are still missing the test files for this pull

I added the required tests.

**Pull request merged**